### PR TITLE
Added API documentation and documentation generation

### PR DESCRIPTION
**Pull request by shmuelk**

### CODE REVIEW

File: Sources/LoggerAPI/Logger.swift [hunk 1]
```diff
@@ -45,62 +57,173 @@ extension LoggerMessageType: CustomStringConvertible {
...
 public class Log {
 
+    /// An instance of the logger. It should usually be the one and only reference
+    /// of the actual `Logger` protocol implementation in the system.
     public static var logger: Logger?
 
-
+    /// Log a log message for use when in verbose logging mode.
+    ///
+    /// - Parameter msg: The mesage to be logged
```
Comment (rob-deans) [inline Sources/LoggerAPI/Logger.swift]: Typo: mesage -> message

File: Sources/LoggerAPI/Logger.swift [hunk 2]
```diff
@@ -45,62 +57,173 @@ extension LoggerMessageType: CustomStringConvertible {
...
     public class func exit(_ msg: String, functionName: String = #function,
         lineNum: Int = #line, fileName: String = #file) {
             logger?.log(.exit, msg: msg,
                 functionName: functionName, lineNum: lineNum, fileName: fileName)
     }
     
+    /// A function that will indicate if a message with a specified type (`LoggerMessageType`)
+    /// will be outputed in the log (i.e. will not be filtered out).
+    ///
+    /// -Parameter type: The type of message that one wants to know if it will be output in the log.
```
Comment (rob-deans) [inline Sources/LoggerAPI/Logger.swift]: Typo: -Parameter -> - Parameter

File: Sources/LoggerAPI/Logger.swift [hunk 3]
```diff
@@ -45,62 +57,173 @@ extension LoggerMessageType: CustomStringConvertible {
...
+    ///                      logger API. Defaults to the file of the actual function
+    ///                      invoking this function.
     public class func exit(_ msg: String, functionName: String = #function,
         lineNum: Int = #line, fileName: String = #file) {
             logger?.log(.exit, msg: msg,
                 functionName: functionName, lineNum: lineNum, fileName: fileName)
     }
     
+    /// A function that will indicate if a message with a specified type (`LoggerMessageType`)
+    /// will be outputed in the log (i.e. will not be filtered out).
```
Comment (rob-deans) [inline Sources/LoggerAPI/Logger.swift]: Typo: outputed -> outputted

File: Sources/LoggerAPI/Logger.swift [hunk 4]
```diff
@@ -45,62 +57,173 @@ extension LoggerMessageType: CustomStringConvertible {
...
     func log(_ type: LoggerMessageType, msg: String,
         functionName: String, lineNum: Int, fileName: String)
     
+    /// A function that will indicate if a message with a specified type (`LoggerMessageType`)
+    /// will be outputed in the log (i.e. will not be filtered out).
+    ///
+    /// -Parameter type: The type of message that one wants to know if it will be output in the log.
+    ///
+    /// - Returns: A Bool indicating whether, if true, or not a message of the specified type
+    ///           (`LoggerMessageType`) would be output.
```
Comment (rob-deans) [inline Sources/LoggerAPI/Logger.swift]: Suggestion: would be output -> will be outputted

Comment (rob-deans) [reply Sources/LoggerAPI/Logger.swift]: Also change where it was copied to

File: Sources/LoggerAPI/Logger.swift (hunk 1 above)
Comment (rob-deans) [reply Sources/LoggerAPI/Logger.swift]: Also change where it was copied to

File: Sources/LoggerAPI/Logger.swift (hunk 4 above)
Comment (shmuelk) [reply Sources/LoggerAPI/Logger.swift]: Done.

File: Sources/LoggerAPI/Logger.swift (hunk 1 above)
Comment (shmuelk) [reply Sources/LoggerAPI/Logger.swift]: Done, all over the place.

File: Sources/LoggerAPI/Logger.swift (hunk 3 above)
Comment (shmuelk) [reply Sources/LoggerAPI/Logger.swift]: Done.

File: Sources/LoggerAPI/Logger.swift (hunk 2 above)
Comment (shmuelk) [reply Sources/LoggerAPI/Logger.swift]: Fixed.

**Pull request merged**